core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to extract image filenames in a State.
+ */
+
+oppia.factory('ExtractImageFilenamesFromStateService', ['HtmlEscaperService',
+  function(HtmlEscaperService) {
+    var INTERACTION_TYPE_MULTIPLE_CHOICE = 'MultipleChoiceInput';
+    var INTERACTION_TYPE_ITEM_SELECTION = 'ItemSelectionInput';
+    var INTERACTION_TYPE_IMAGE_CLICK_INPUT = 'ImageClickInput';
+
+    var filenamesInState = [];
+    var _getAllHtmlOfState = function(state) {
+      var _allHtmlInTheState = [];
+      // The order of the extracted image names is same as they appear in a
+      // state. The images should be preloaded in the following order ---
+      // content, customizationArgs of interactions, feedback of outcomes ()
+      // including feedback of default outcome if any), hints, solution if any.
+      var stateContentHtml = state.content.getHtml();
+      _allHtmlInTheState.push(stateContentHtml);
+
+      if (state.interaction.id === INTERACTION_TYPE_MULTIPLE_CHOICE ||
+          state.interaction.id === INTERACTION_TYPE_ITEM_SELECTION ) {
+        var customizationArgsHtml = '';
+        state.interaction.customizationArgs.choices.value.forEach(
+          function(value) {
+            customizationArgsHtml = customizationArgsHtml.concat(value);
+          });
+        _allHtmlInTheState.push(customizationArgsHtml);
+      }
+
+      state.interaction.answerGroups.forEach(function(answerGroup) {
+        var answerGroupHtml = answerGroup.outcome.feedback.getHtml();
+        _allHtmlInTheState.push(answerGroupHtml);
+      });
+
+      if (state.interaction.defaultOutcome !== null) {
+        var defaultOutcomeHtml =
```

Comment:
Add `= (` (so we are consistent with Python)

Reply:
Done!